features/poller/monitors/core/src/main/java/org/opennms/netmgt/poller/monitors/support/ParameterSubstitutingMonitor.java

```diff
@@ -0,0 +1,177 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2018 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2018 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.poller.monitors.support;
+
+import java.beans.BeanInfo;
+import java.beans.Introspector;
+import java.beans.IntrospectionException;
+import java.beans.PropertyDescriptor;
+import java.io.UnsupportedEncodingException;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+
+import org.opennms.core.spring.BeanUtils;
+import org.opennms.core.utils.ParameterMap;
+import org.opennms.netmgt.dao.api.NodeDao;
+import org.opennms.netmgt.model.OnmsNode;
+import org.opennms.netmgt.model.OnmsAssetRecord;
+import org.opennms.netmgt.poller.MonitoredService;
+import org.opennms.netmgt.poller.support.AbstractServiceMonitor;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.beans.BeanWrapper;
+import org.springframework.beans.BeanWrapperImpl;
+
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+/**
+ * {@Inheritdoc}
+ * Uses {@link org.opennms.netmgt.poller.ServiceMonitor#getRuntimeAttributes(MonitoredService svc, Map<String, Object> parameters)}
+ * to add parameters with substitutions of placeholders in existing parameters.
+ * The new parameter keys will be prepended with <code>'subbed-'</code>.
+ *
+ * @author <A HREF="dschlenk@convergeone.com">David Schlenk</A>
+ */
+public abstract class ParameterSubstitutingMonitor extends AbstractServiceMonitor {
+
+    public static final Logger LOG = LoggerFactory.getLogger(ParameterSubstitutingMonitor.class);
+
+    private static Map<String, Pattern> subPatterns = new HashMap<>();
```

Comment:
Can we either use the non `m_`prefix style or the `m_` prefix style, but do not mix both?